File: lsd-interceptors/src/main/java/io/lsdconsulting/interceptors/common/AppName.java
package io.lsdconsulting.interceptors.common;

import com.lsd.core.domain.ComponentName;
import lombok.Value;

@Value
public class AppName {
    String value;

    public AppName(String value) {
        this.value = new ComponentName(value).getName();
    }
}

File: lsd-interceptors/src/main/java/io/lsdconsulting/interceptors/common/AppName.kt
package io.lsdconsulting.interceptors.common

import com.lsd.core.domain.ComponentName

data class AppName private constructor(val value: String) {

    companion object Factory {
        fun create(value: String) : AppName {
            return AppName(ComponentName(value).normalisedName)
        }
    }
}
